e2e/suite_test.go

```diff
@@ -65,32 +57,23 @@ func maybeSkipScenario(t *testing.T, s *scenario.Scenario) {
 	t.Logf("running scenario %q with image %q", s.Name, rid)
 }
 
-func setupAndRunScenario(ctx context.Context, t *testing.T, e2eScenario *scenario.Scenario, r *mrand.Rand, clusterConfigs []clusterConfig) {
-	clusterConfig, err := chooseCluster(ctx, r, e2eScenario, clusterConfigs)
-	if err != nil {
-		t.Fatal(err)
-	}
+func setupAndRunScenario(ctx context.Context, t *testing.T, e2eScenario *scenario.Scenario, clusterConfig *cluster.Cluster) {
+	log.Printf("chose cluster: %q", *clusterConfig.Model.ID)
 
-	clusterName := *clusterConfig.cluster.Name
-	log.Printf("chose cluster: %q", clusterName)
+	clusterParams, err := pollExtractClusterParameters(ctx, clusterConfig.Kube)
```

Comment:
I think we can do this once per-cluster before any scenarios are executed, probably with new fields on `cluster.Cluster`

Reply:
it's actually requires moving a lot of code between packages. I have some ideas how to improve it.
But I don't want to add 2k LOC diff to the PR.

Is it ok if I optimize it later?